sanic/worker.py

```diff
@@ -12,199 +12,207 @@
 
 try:
     import uvloop
+    import gunicorn.workers.base as base
     asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
 except ImportError:
+    base = None
     pass
-import gunicorn.workers.base as base
 
 from sanic.server import trigger_events, serve, HttpProtocol, Signal
 from sanic.websocket import WebSocketProtocol
 
 
-class GunicornWorker(base.Worker):
-
-    http_protocol = HttpProtocol
-    websocket_protocol = WebSocketProtocol
-
-    def __init__(self, *args, **kw):  # pragma: no cover
-        super().__init__(*args, **kw)
-        cfg = self.cfg
-        if cfg.is_ssl:
-            self.ssl_context = self._create_ssl_context(cfg)
-        else:
-            self.ssl_context = None
-        self.servers = {}
-        self.connections = set()
-        self.exit_code = 0
-        self.signal = Signal()
-
-    def init_process(self):
-        # create new event_loop after fork
-        asyncio.get_event_loop().close()
-
-        self.loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(self.loop)
-
-        super().init_process()
-
-    def run(self):
-        is_debug = self.log.loglevel == logging.DEBUG
-        protocol = (
-            self.websocket_protocol if self.app.callable.websocket_enabled
-            else self.http_protocol)
-        self._server_settings = self.app.callable._helper(
-            loop=self.loop,
-            debug=is_debug,
-            protocol=protocol,
-            ssl=self.ssl_context,
-            run_async=True)
-        self._server_settings['signal'] = self.signal
-        self._server_settings.pop('sock')
-        trigger_events(self._server_settings.get('before_start', []),
-                       self.loop)
-        self._server_settings['before_start'] = ()
-
-        self._runner = asyncio.ensure_future(self._run(), loop=self.loop)
-        try:
-            self.loop.run_until_complete(self._runner)
-            self.app.callable.is_running = True
-            trigger_events(self._server_settings.get('after_start', []),
+# gunicorn is not available on windows
+if base is not None:
```

Comment:
This is actually unnecessary.

Gunicorn doesn't support windows, and this module requires Gunicorn. I don't think users on windows want to import this worker module.

Reply:
Also note, gunicorn is not a direct dependency of Sanic.

Reply:
workers seemed like a generic enough idea that I left it but you may be right, should I just let the import error raise when workers tries to import gunicorn then?

Reply:
yes, since the worker module is useless on windows.